rdr_service/dao/genomics_dao.py

```diff
@@ -3245,3 +3256,120 @@ def get_w3sr_records(self, **kwargs):
+            ).join(
+                BiobankOrderIdentifier,
+                BiobankOrderIdentifier.value == BiobankStoredSample.biobankOrderIdentifier
+            ).join(
+                BiobankOrder,
+                BiobankOrder.biobankOrderId == BiobankOrderIdentifier.biobankOrderId
+            ).join(
+                sample_collected_site,
+                sample_collected_site.siteId == BiobankOrder.collectedSiteId
+            ).join(
+                informing_loop_subquery,
+                informing_loop_subquery.participant_id == GenomicSetMember.participantId
+            ).outerjoin(
+                GenomicResultWorkflowState,
+                and_(
+                    GenomicResultWorkflowState.genomic_set_member_id == GenomicSetMember.id,
+                    GenomicResultWorkflowState.results_module == result_module_type
+                )
+            ).filter(
+                GenomicGCValidationMetrics.processingStatus.like('pass'),
+                GenomicSetMember.genomeType == config.GENOME_TYPE_WGS,
+                ParticipantSummary.withdrawalStatus == WithdrawalStatus.NOT_WITHDRAWN,
+                ParticipantSummary.suspensionStatus == SuspensionStatus.NOT_SUSPENDED,
+                ParticipantSummary.deceasedStatus == DeceasedStatus.UNSET,
+                GenomicGCValidationMetrics.sexConcordance.like('true'),     # check AW2 gives sex concordance as true
+                GenomicGCValidationMetrics.drcSexConcordance.like('pass'),  # check AW4 gives sex concordance as true
+                GenomicSetMember.qcStatus == GenomicQcStatus.PASS,
+                GenomicSetMember.gcManifestSampleSource.like('whole blood'),
+                ParticipantSummary.consentForStudyEnrollment == QuestionnaireStatus.SUBMITTED,
+
+                # all data files have been received
+                GenomicGCValidationMetrics.hfVcfReceived == 1,
+                GenomicGCValidationMetrics.hfVcfTbiReceived == 1,
+                GenomicGCValidationMetrics.hfVcfMd5Received == 1,
+                GenomicGCValidationMetrics.cramReceived == 1,
+                GenomicGCValidationMetrics.cramMd5Received == 1,
+                GenomicGCValidationMetrics.craiReceived == 1,
+                GenomicGCValidationMetrics.gvcfReceived == 1,
+                GenomicGCValidationMetrics.gvcfMd5Received == 1,
+
+                ParticipantSummary.consentForGenomicsROR == QuestionnaireStatus.SUBMITTED,
+                GenomicGCValidationMetrics.drcFpConcordance.like('pass'),
+                sample_collected_site.siteType != 'diversion pouch',
+                GenomicSetMember.gcSiteId.like(gc_site_id),
+                ParticipantSummary.participantOrigin != 'careevolution',
+
+                GenomicSetMember.ignoreFlag != 1,
+                GenomicSetMember.genomicWorkflowState == GenomicWorkflowState.CVL_READY,
+                GenomicResultWorkflowState.id.is_(None)
```

Comment:
Can you change this to look for the PGx or HDR W1IL job run ID instead? I know it wasn't clear from the requirements, but I can foresee a future request that may create a case where we have a `GenomicResultWorkflowState` record for a reason upstream and independent of inclusion in the W1IL manifest.

Reply:
Yeah, definitely!